qa/tasks/cephfs/test_volumes.py

```diff
@@ -2869,12 +2869,38 @@ def test_subvolume_info(self):
 
         self.assertEqual(subvol_info["earmark"], earmark)
         
+        self.assertEqual(subvol_info['source']['volume'], 'N/A')
+        self.assertEqual(subvol_info['source']['subvolume'], 'N/A')
+        self.assertEqual(subvol_info['source']['snapshot'], 'N/A')
+        self.assertEqual(subvol_info['source']['group'], 'N/A')
+
         # remove subvolumes
         self._fs_cmd("subvolume", "rm", self.volname, subvolume)
 
         # verify trash dir is clean
         self._wait_for_trash_empty()
 
+    def test_subvol_src_info_with_custom_group(self):
+        '''
+        Test that source subvolume's group is printed as N/A when
+        "subvolume info" command is run for subvolume that wasn't created
+        through cloning.
+        '''
+        subvol_name = self._gen_subvol_name()
+        group_name = self._gen_subvol_grp_name()
+
+        self.run_ceph_cmd(f'fs subvolumegroup create {self.volname} '
+                          f'{group_name}')
+        self.run_ceph_cmd(f'fs subvolume create {self.volname} {subvol_name} '
+                          f'{group_name}')
+
+        subvol_info = self.get_ceph_cmd_stdout(
+            f'fs subvolume info {self.volname} {subvol_name} {group_name}')
+        subvol_info = json.loads(subvol_info)
+
+        # rest of the fields are checked for in test_subvolume_info.
+        self.assertEqual(subvol_info['source']['group'], 'N/A')
```

Comment:
If the subvolume is not a clone, does it even make sense to have source information?

Reply:
That's the current state of code -- source field exists in `subvolume info` command only if subvolume is a clone.